source/common/http/conn_pool_base.h

```diff
@@ -48,6 +48,7 @@ class ConnPoolImplBase : protected Logger::Loggable<Logger::Id::pool> {
   const Upstream::HostConstSharedPtr host_;
   const Upstream::ResourcePriority priority_;
   std::list<PendingRequestPtr> pending_requests_;
+  std::list<PendingRequestPtr> pending_requests_to_purge_;
```

Comment:
Can you add comments here and where the fix is? That will help me wrap my head around your thinking and we can go from there.